addon/usables/use-machine.js

```diff
@@ -0,0 +1,102 @@
+import { interpret } from 'xstate';
+import { setUsableManager } from 'ember-usable';
+import { later, cancel } from '@ember/runloop';
+import { tracked } from '@glimmer/tracking';
+import { getOwner, setOwner } from '@ember/application';
+
+export class InterpreterService {
+  @tracked service;
+  @tracked currentState;
+
+  constructor(machine) {
+    this.machine = machine;
+  }
+
+  get state() {
+    return {
+      currentState: this.currentState,
+      send: this.service.send,
+    };
+  }
+
+  setup() {
+    this.service = interpret(this.machine, {
+      clock: {
```

Comment:
why is this clock here? / what's it for?

Reply:
https://xstate.js.org/api/classes/interpreter.html#clock

> The clock that is responsible for setting and clearing timeouts, such as delayed events and transitions.

We are making sure that we use the ember runloop when the xstate-interpreter schedules delays, transitions etc.. This is important when we are testing etc.